common/beerocks/bcl/include/bcl/beerocks_promise.h

```diff
@@ -22,53 +23,70 @@ template <class T> class promise {
     // Set value for waiting threads
     void set_value(const T value)
     {
-        pthread_mutex_lock(&m_mut);
+        int err;
+        if ((err = pthread_mutex_lock(&m_mut)) != 0) {
+            LOG(ERROR) << "pthread_mutex_lock failed, error code: " << err;
+            return;
+        }
         m_value  = value;
         m_signal = 1;
-        pthread_cond_broadcast(&m_cond);
-        pthread_mutex_unlock(&m_mut);
+        if ((err = pthread_cond_broadcast(&m_cond)) != 0)
+            LOG(ERROR) << "pthread_cond_broadcast failed, error code: " << err;
```

Comment:
Is there a point to do the next check if this one fails?

Reply:
Yes, the mutex must still be unlocked if it fails.

But actually, this is the reason why I think it should be asserts: if this fails, we have broken invariants and there is no way to recover. Most likely something surprising will go wrong at some (potentially distant) point in the future. IMO not asserting here is even a security issue, because it might be used by an attacker to trigger specific, undesired behaviour.

Reply:
So what's blocking us from adding asserts?

Reply:
I'll add it now.